app/src/main/java/com/marat/hvatit/playlistmaker2/SearchActivity.kt

```diff
@@ -2,78 +2,67 @@ package com.marat.hvatit.playlistmaker2
 import com.marat.hvatit.playlistmaker2.adapters.TrackListAdapter
 import com.marat.hvatit.playlistmaker2.models.Track
-import com.bumptech.glide.Glide
+import com.google.gson.Gson
+import com.marat.hvatit.playlistmaker2.service.AppleMusicAPI
+import com.marat.hvatit.playlistmaker2.service.AppleSong
+import com.marat.hvatit.playlistmaker2.service.AppleSongResponce
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.Query
 
 const val EDITTEXT_TEXT = "EDITTEXT_TEXT"
 private const val TAG = "SearchActivity"
+private lateinit var disconnected: String
+private lateinit var nothingToShow: String
+private lateinit var allfine: String
+
+enum class SearchActivityState(val value: String) {
+    DISCONNECTED(""), NOTHINGTOSHOW(""), ALLFINE("")
+}
 
 class SearchActivity : AppCompatActivity() {
 
     private var saveEditText: String = "error"
 
-    companion object {
+    private val appleBaseUrl = "https://itunes.apple.com"
 
-        val data: ArrayList<Track> = initialize()
-
-        private fun initialize(): ArrayList<Track> {
-            val songList: ArrayList<Track> = arrayListOf()
-            songList.add(
-                Track(
-                    "Smells Like Teen Spirit",
-                    "Nirvana",
-                    "5:01",
-                    "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/7b/58/c2/7b58c21a-2b51-2bb2-e59a-9bb9b96ad8c3/00602567924166.rgb.jpg/100x100bb.jpg"
-                )
-            )
-            songList.add(
-                Track(
-                    "Billie Jean",
-                    "Michael Jackson",
-                    "4:35",
-                    "https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/3d/9d/38/3d9d3811-71f0-3a0e-1ada-3004e56ff852/827969428726.jpg/100x100bb.jpg"
-                )
-            )
+    private val retrofit =
+        Retrofit.Builder().baseUrl(appleBaseUrl).addConverterFactory(GsonConverterFactory.create())
+            .build()
 
-            songList.add(
-                Track(
-                    "Stayin' Alive",
-                    "Bee Gees",
-                    "4:10",
-                    "https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/1f/80/1f/1f801fc1-8c0f-ea3e-d3e5-387c6619619e/16UMGIM86640.rgb.jpg/100x100bb.jpg"
-                )
-            )
+    private val appleService = retrofit.create(AppleMusicAPI::class.java)
 
-            songList.add(
-                Track(
-                    "Whole Lotta Love",
-                    "Led Zeppelin",
-                    "5:33",
-                    "https://is2-ssl.mzstatic.com/image/thumb/Music62/v4/7e/17/e3/7e17e33f-2efa-2a36-e916-7f808576cf6b/mzm.fyigqcbs.jpg/100x100bb.jpg"
-                )
-            )
+    private var appleSongList = ArrayList<AppleSong>()
+    private var trackListAdapter = TrackListAdapter(appleSongList)
```

Comment:
```suggestion
    private val appleSongList = ArrayList<AppleSong>()
    private val trackListAdapter = TrackListAdapter(appleSongList)
```
Список уже изменяемый, сами переменные изменяемыми делать не нужно